hub-js/graphql/local/schema.graphql

```diff
@@ -503,7 +503,9 @@ type Mutation {
           RETURN specBackend
         ',
         '
-          CREATE (specBackend: TypeInstanceResourceVersionSpecBackend {context: apoc.convert.toJson(item.typeInstance.backend.context)})
+          MATCH (latestRevision)-[:SPECIFIED_BY]->(latestSpec: TypeInstanceResourceVersionSpec)
```

Comment:
While it works for:

✅  Case 1: Set context to null
```graphql
mutation UpdateTypeInstances($typeInstanceId: ID!) {
  updateTypeInstances(
    in: [
      {
        ownerID: "namespace/name"
        id: $typeInstanceId
        typeInstance: {
          attributes: [{ path: "cap.attribute.sample", revision: "0.1.0" }]
          value: { name: "Luke Skywalker3" }
          backend: {
            context: null 
          }
        }
      }
    ]
  ) {
    id
    
  }
}
```
and

✅  Case 2: Do not provide backend - reuse old context
```graphql
mutation UpdateTypeInstances($typeInstanceId: ID!) {
  updateTypeInstances(
    in: [
      {
        ownerID: "namespace/name"
        id: $typeInstanceId
        typeInstance: {
          attributes: [{ path: "cap.attribute.sample", revision: "0.1.0" }]
          value: { name: "Luke Skywalker3" }
         # no backend at all - reuse old context
        }
      }
    ]
  ) {
    id
    
  }
}

```

I think it should also work for:

❌  Case 3: Provide backend object, but without specifying `context` field:

```graphql
mutation UpdateTypeInstances($typeInstanceId: ID!) {
  updateTypeInstances(
    in: [
      {
        ownerID: "namespace/name"
        id: $typeInstanceId
        typeInstance: {
          attributes: [{ path: "cap.attribute.sample", revision: "0.1.0" }]
          value: { name: "Luke Skywalker3" }
          backend: { 
            # context: {
            #  provider: "foo"
            # }
          }
        }
      }
    ]
  ) {
    id
    
  }
}
```

But it doesn't - it makes the context `null`. It would make sense, cause there might be more fields inside `backend` object, and if I don't want to change `context`, I just don't provide it.

In JS it is easy to check whether a given property is `null` or `undefined` (not provided by user). Not sure if we can achieve this in Cypher though 🤔  If that's not easy, let's skip it 🙂

Reply:
Yep, there is a test that showcase this scenario, unfortunately AFAIK it's not possible with Cypher (https://neo4j.com/developer/kb/understanding-non-existent-properties-and-null-values/). Probably this will need to be done on the JS side but we use also `cypherMutation` which generates this query and due to that fact it's not easy. 

As we don't have any other field under `backend` I will go with what you proposed - skip it.
We can revisit that when a new property will be added,  and figure out how to handle that nicely. But IMO it won't be done in the near future 😄